### ISSUE TITLE
Add ability to over-ride a jobs quiet period

### ISSUE THREAD

**Issue by twinderw** [jenkinsci/generic-webhook-trigger-plugin#159]
Hi,

Currently the plugin respects the triggered job's quiet period, which is great. However I have a use case where it would be better to be able to override the job's quiet period if we wish.

The idea is that something like "curl -vs http://localhost:8080/jenkins/generic-webhook-trigger/invoke?token=abc123?quietperiod=5" would result in the quiet period being set to 5 minutes regardless of what the job or the global settings indicate.

Taking a look at the plugin's code it seems like this should be pretty easy to implement, and I'm willing to do the work myself. Before I put the time in though, I just wanted to get some feedback on if this feature would be approved.

Thanks!
Tom

**Comment by tomasbjerre** [comment]
My only concern is that someone may use that parameter name, quietperiod, as input already. Perhaps jobquietperiod would reduce that risk.

Feel free to open a PR.

**Comment by tomasbjerre** [comment]
Another concern might be users who do not wish to allow an invoker to set quietperiod. That might be a bigger issue... Perhaps you should wait with working on a pr.

I would like to investigate if one could use a variable in the quietperiod configuration. And resolve that variable with this plugin. Perhaps that is a better solution and perhaps that requiers development in Jenkins core.

**Comment by twinderw** [comment]
Ok I'll hold off on starting work, please let me know what solution you come up with and what my part in that will be.

**Comment by twinderw** [comment]
Would it be enough to add an option in the plugin's configuration "Allow trigger to override quiet period" or something similar?

**Comment by tomasbjerre** [comment]
Yes that would probably be enough.

**Comment by twinderw** [comment]
Excellent, I will work on this later this week.

**Comment by twinderw** [comment]
@tomasbjerre Forgive me if this is a basic question, I'm new to jenkins plugin development. I've got my code done, but I'm trying to load a jenkins instance with the update to the plugin and I'm having trouble. I've found instructions online to run mvn hpi:run which should load a jenkins instance for me, but it's failing with the following error: 
"Failed to execute goal org.jenkins-ci.tools:maven-hpi-plugin:2.1:run (default-cli) on project generic-webhook-trigger: Failure: Address already in use: bind"
Have you run into this before? Or maybe is there a better way for me to test my changes?
